sim/gui.py

```diff
@@ -293,6 +281,27 @@ def draw_maze(
             if Walls.WEST in walls:
                 pg.draw.line(self.screen, wall_color, (x, y - line_end), (x, y + line_end + tile_size), self.wall_thickness)
 
+        # Draw robot's route after all other maze elements:
+        for i, cell in enumerate(maze.route):
+            x1 = cell[1] * tile_size + offset[0]
+            y1 = cell[0] * tile_size + offset[1]
+            if i + 1 < len(maze.route):
+                next_cell = maze.route[i + 1]
+                x2 = next_cell[1] * tile_size + offset[0]
+                y2 = next_cell[0] * tile_size + offset[1]
+                self.draw_route_cell(Position(x1, y1), Position(x2, y2), 'magenta')
+
+        self.draw_robot(robot_position, robot_direction)
+        for row, col, extra_info in maze.iter_info():
+            x = col * tile_size + offset[0]
+            y = row * tile_size + offset[1]
+            current_pos = Position(x, y)
+            if extra_info.weight is not None:
+                self.draw_text_by_center(
+                    str(extra_info.weight),
+                    self.half_tile,
+                    current_pos + Position(self.half_tile + 1, self.half_tile + 1))
```

Comment:
Improper parameter break, use:
```suggestion
                    current_pos + Position(self.half_tile + 1, self.half_tile + 1),
                )
```